pkg/cmd/roachtest/operations/probe_ranges.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package operations
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operation"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestflags"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+)
+
+// runProbeRanges runs a read or write stmt against crdb_internal.probe_ranges.
+func runProbeRanges(
+	ctx context.Context, o operation.Operation, c cluster.Cluster, writeProbe bool,
+) registry.OperationCleanup {
+	rng, _ := randutil.NewPseudoRand()
```

Comment:
an aside: it doesn't seem ideal for operations to be managing their own RNG. e.g. if we want to reproduce a specific operation run, it's up to each individual operation to make sure to log the seed which we aren't doing here.

Instead, the top level operation runner should be calling `randutil.NewPseudoRand()` which uses that to generate+log a new seed for each individual operation.

Anyway, outside the scope of this PR, just mentioning it.